### ISSUE TITLE
Double/Multiple of the same log line are sent to Cloudwatch

### ISSUE THREAD

**Issue by royb-tabit** [aws/aws-for-fluent-bit#213]
Using Helm Chart 1.7.0
and these versions: 2.16.1/2.18.0
While making sure there are no double logs on with "kubectl log ..."

Cloudwatch is systematically getting 2/3/4 copies of the same line.

Using this ConfigMap:

`data:
  fluent-bit.conf: |
    [SERVICE]
        Parsers_File /fluent-bit/parsers/parsers.conf

    [INPUT]
        Name              tail
        Tag               kube.*
        Path              /var/log/containers/*.log
        DB                /var/log/flb_kube.db
        Parser            docker
        Docker_Mode       On
        Mem_Buf_Limit     5MB
        Skip_Long_Lines   On
        Refresh_Interval  10

    [FILTER]
        Name                kubernetes
        Match               kube.*
        Kube_URL            https://kubernetes.default.svc.cluster.local:443
        Merge_Log           On
        Merge_Log_Key       data
        Keep_Log            On
        K8S-Logging.Parser  On
        K8S-Logging.Exclude On
    [OUTPUT]
        Name                  cloudwatch
        Match                 *
        region                eu-west-1
        log_group_name        eks-il-int
        log_stream_prefix     fluentbit-
        auto_create_group     true
`

**Comment by zhonghui12** [comment]
hi @royb-tabit, could you please show us some log information/output of it? Thanks

**Comment by royb-tabit** [comment]
hi @zhonghui12 , sorry for the late response,
here is a Typical log (other pods looks the same), no errors:

```
Fluent Bit v1.8.2
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/08/12 07:34:54] [ info] [engine] started (pid=1)
[2021/08/12 07:34:54] [ info] [storage] version=1.1.1, initializing...
[2021/08/12 07:34:54] [ info] [storage] in-memory
[2021/08/12 07:34:54] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2021/08/12 07:34:54] [ info] [cmetrics] version=0.1.5
[2021/08/12 07:34:54] [ info] [filter:kubernetes:kubernetes.0] https=1 host=kubernetes.default.svc.cluster.local port=443
[2021/08/12 07:34:54] [ info] [filter:kubernetes:kubernetes.0] local POD info OK
[2021/08/12 07:34:54] [ info] [filter:kubernetes:kubernetes.0] testing connectivity with API server...
[2021/08/12 07:34:54] [ info] [filter:kubernetes:kubernetes.0] connectivity OK
AWS for Fluent Bit Container Image Version 2.18.0time="2021-08-12T07:34:54Z" level=info msg="[cloudwatch 0] plugin parameter log_group_name = 'eks-il-int'"
time="2021-08-12T07:34:54Z" level=info msg="[cloudwatch 0] plugin parameter default_log_group_name = 'fluentbit-default'"
time="2021-08-12T07:34:54Z" level=info msg="[cloudwatch 0] plugin parameter log_stream_prefix = 'fluentbit-'"
time="2021-08-12T07:34:54Z" level=info msg="[cloudwatch 0] plugin parameter log_stream_name = ''"
time="2021-08-12T07:34:54Z" level=info msg="[cloudwatch 0] plugin parameter default_log_stream_name = '/fluentbit-default'"
time="2021-08-12T07:34:54Z" level=info msg="[cloudwatch 0] plugin parameter region = 'eu-west-1'"
time="2021-08-12T07:34:54Z" level=info msg="[cloudwatch 0] plugin parameter log_key = ''"
time="2021-08-12T07:34:54Z" level=info msg="[cloudwatch 0] plugin parameter role_arn = ''"
time="2021-08-12T07:34:54Z" level=info msg="[cloudwatch 0] plugin parameter auto_create_group = 'true'"
time="2021-08-12T07:34:54Z" level=info msg="[cloudwatch 0] plugin parameter new_log_group_tags = ''"
time="2021-08-12T07:34:54Z" level=info msg="[cloudwatch 0] plugin parameter log_retention_days = '0'"
time="2021-08-12T07:34:54Z" level=info msg="[cloudwatch 0] plugin parameter endpoint = ''"
time="2021-08-12T07:34:54Z" level=info msg="[cloudwatch 0] plugin parameter sts_endpoint = ''"
time="2021-08-12T07:34:54Z" level=info msg="[cloudwatch 0] plugin parameter credentials_endpoint = "
time="2021-08-12T07:34:54Z" level=info msg="[cloudwatch 0] plugin parameter log_format = ''"
[2021/08/12 07:34:54] [ info] [sp] stream processor started
[2021/08/12 07:34:54] [ info] [input:tail:tail.0] inotify_fs_add(): inode=22036064 watch_fd=1 name=/var/log/containers/aws-node-wbqjz_kube-system_aws-node-cc7f527c698d519186152982da79cf0f4d4a8c84beec28247a4dcd06a2dbd124.log
[2021/08/12 07:34:54] [ info] [input:tail:tail.0] inotify_fs_add(): inode=3550812 watch_fd=2 name=/var/log/containers/aws-node-wbqjz_kube-system_aws-vpc-cni-init-b45bc4088a0a38aa30fbf27cdcad01766c23cd37251cb6523a2eedac9dfd6746.log
[2021/08/12 07:34:54] [ info] [input:tail:tail.0] inotify_fs_add(): inode=36788401 watch_fd=3 name=/var/log/containers/cloner-monitor-7c6d9f9f67-wwp7f_applications_cloner-monitor-887dd8157199402fb1c6fa3c72b99545f9cea6cd4b9410977ed3f3c50d33d8cf.log
[2021/08/12 07:34:54] [ info] [input:tail:tail.0] inotify_fs_add(): inode=34677177 watch_fd=4 name=/var/log/containers/coredns-6dc98fd64c-6jndg_kube-system_coredns-6e449d9c1da3626bbc6018b11173a54450af378edc3d1d568f68f7d18dc7aed1.log
[2021/08/12 07:34:54] [ info] [input:tail:tail.0] inotify_fs_add(): inode=252480 watch_fd=5 name=/var/log/containers/kube-proxy-hb4vn_kube-system_kube-proxy-6ec7f718bebb852b296faf174e58071e4eed840e1aaedff0edaec7a9abedee8f.log
[2021/08/12 07:34:54] [ info] [input:tail:tail.0] inotify_fs_add(): inode=11580394 watch_fd=6 name=/var/log/containers/newrelic-bundle-newrelic-infrastructure-88q5v_newrelic_newrelic-infrastructure-33d4861d9e81fead49f4f7ff7dc9d7da4c5c20a728d2fdd6601126511924afbb.log
[2021/08/12 07:34:54] [ info] [input:tail:tail.0] inotify_fs_add(): inode=5250254 watch_fd=7 name=/var/log/containers/newrelic-bundle-newrelic-logging-qj7vn_newrelic_newrelic-logging-4234e7bcf3cd19f3395dfd2c937d01b356c946df8155e2a3d7b0aac4a40b2dbe.log
[2021/08/12 07:34:54] [ info] [input:tail:tail.0] inotify_fs_add(): inode=301797 watch_fd=8 name=/var/log/containers/aws-for-fluent-bit-vsrt4_kube-system_aws-for-fluent-bit-4c513ad1a480d1ca8e863fbcb39f9d6c4036b70baea7cc458c00273dd43fa465.log
time="2021-08-12T07:35:00Z" level=info msg="[cloudwatch 0] Log group eks-il-int already exists\n"
time="2021-08-12T07:35:00Z" level=info msg="[cloudwatch 0] Created log stream fluentbit-kube.var.log.containers.aws-for-fluent-bit-vsrt4_kube-system_aws-for-fluent-bit-4c513ad1a480d1ca8e863fbcb39f9d6c4036b70baea7cc458c00273dd43fa465.log in group eks-il-int"
```

**Comment by hossain-rayhan** [comment]
Hi @royb-tabit, thanks for the log. However, some details with dubug log will help us. Can you please enable `Log_Level=debug` and share some more logs. 

I am also interested to know your payload size. If you are sending too many logs within a short period of time and fluent-bit is getting throttled by CloudWatch and retrying, it may publish duplicate logs. This is kind of expected behavior. However, more debug log will help us for better investigation.

**Comment by matthewfala** [comment]
Closing for no response. Please reopen if you have an update. Thanks!

**Comment by ek8484** [comment]
I had a similar duplicate log records issue recently. And the `[OUTPUT]` section configuration was similar.
During troubleshooting I discovered that logs from containers have been sent not only to a separate stream per container, but also in some general stream with name `fluentbit-kube` from each container.
I changed `Match` field configuration from `*` to `kube.*` and this helped to stop sending logs to that general stream.
```
[OUTPUT]
    Name                  cloudwatch
    Match                 kube.*
    region                eu-west-1
    log_group_name        eks-il-int
    log_stream_prefix     fluentbit-
    auto_create_group     true
```
